fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneTableFunctionColumnRule.java

```diff
@@ -40,21 +42,61 @@ public PruneTableFunctionColumnRule() {
     public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
         LogicalTableFunctionOperator logicalTableFunctionOperator = (LogicalTableFunctionOperator) input.getOp();
         ColumnRefSet requiredOutputColumns = context.getTaskContext().getRequiredColumns();
-
         List<ColumnRefOperator> newOuterCols = Lists.newArrayList();
+
         for (ColumnRefOperator col : logicalTableFunctionOperator.getOuterColRefs()) {
             if (requiredOutputColumns.contains(col.getId())) {
```

Comment:
this may prune all cloumns.

Reply:
I have considered this point. unnest function supports empty columns.

```
class Unnest final : public TableFunction {
public:
    std::pair<Columns, UInt32Column::Ptr> process(TableFunctionState* state) const override {
        if (state->get_columns().empty()) {
            return {};
        }
```